gateways/stripe/do_transaction.php

```diff
@@ -12,40 +12,55 @@ function espresso_process_stripe($payment_data) {
+		//Alternative Stripe settings set on the event, set the key to be used here.
+    	$stripe_settings['stripe_secret_key'] = $payment_data['event_meta']['stripe_secret_key'];
+	}
+
+	//Include the Stripe API
+	require_once (dirname(__FILE__).'/stripe-php-1.18.0/lib/Stripe.php');
+
+	//Pull the stripeToken posted by Checkout.
+	$token = $_POST['stripeToken'];
 
-	$cc = $_POST['cc'];
-	$exp_month = $_POST['exp_month'];
-	$exp_year = $_POST['exp_year'];
-	$csc = $_POST['csc'];
-	$bname = $_POST['first_name'] . " " . $_POST['last_name'];
-	$baddress = $_POST['address'];
-	$bcity = $_POST['city'];
-	$bzip = $_POST['zip'];
-	$email = $_POST['email'];
-	$line_item = "LINEITEM~PRODUCTID=" . $payment_data['attendee_id'] . "+DESCRIPTION=" . $payment_data["event_name"] . "[" . date('m-d-Y', strtotime($payment_data['start_date'])) . "]" . " >> " . $payment_data["fname"] . " " . $payment_data["lname"] . "
-							QUANTITY=1 UNITCOST=" . $payment_data['total_cost'];
+	//Set the Stripe API secret key.
+	Stripe::setApiKey( $stripe_settings['stripe_secret_key'] );
+    
+    //Build the Stripe data array
+    $stripe_data = array(
+        'amount' => str_replace( array(',', '.'), '', number_format( $payment_data['total_cost'], 2) ),
+        'currency' => $stripe_settings['stripe_currency_symbol'],
+        'card' => $token,
+        'description' => $payment_data["event_name"] . 
+        				"[" . date('m-d-Y', strtotime($payment_data['start_date'])) . "]" . 
+        				" >> " . $payment_data["fname"] . " " . $payment_data["lname"]
+    );
+	
+	//Create a charge with Stripe.
+	$charge = Stripe_Charge::create( $stripe_data );
 
-	$response = $cls_stripe->do_transaction($payment_data['total_cost'], $cc, $csc, $exp_month, $exp_year, $bname, $line_item, $payment_data);
-	if (!empty($response)) {
-		$payment_data['txn_details'] = serialize($response);
-		if (isset($response['status'])) {
+	//Convert the Stripe charge to an array so we can use it.
+	$charge_array = $charge->__toArray(true);
+	
+	if ( !empty( $charge_array ) ) {
+		$payment_data['txn_details'] = serialize( $charge_array );
+		if ( isset($charge_array['status'] ) ) {
 			echo "<div id='stripe_response'>";
-			if ($response['status'] > 0) {
-				echo "<div class='stripe_status'>" . $response['msg'] . "</div>";
+			if ($charge_array['paid'] === TRUE) {
+				echo "<div class='stripe_status'>" . $charge_array['outcome']['seller_message'] . "</div>";
 				$payment_data['payment_status'] = 'Completed';
-				$payment_data['txn_id'] = $response['txid'];
+				$payment_data['txn_id'] = $charge_array['id'];
 			}
-			if (isset($response['error_msg']) && strlen(trim($response['error_msg'])) > 0) {
-				echo "<div class='stripe_error'>ERROR: " . $response['error_msg'] . "  </div>";
+			if ( !empty($charge_array['failure_code']) ) {
+				echo "<div class='stripe_error'>ERROR: " . $charge_array['failure_code'] . " - " . $charge_array['failure_message'] . "  </div>";
```

Comment:
I'd be inclined to use a translatable string here, which might especially be good for RTL languages.

Reply:
Done